pkg/querier/remote_read.go

```diff
@@ -333,10 +334,34 @@ func streamChunkedReadResponses(stream io.Writer, ss storage.ChunkSeriesSet, que
 	return ss.Err()
 }
 
-func initializedFrameBytesRemaining(maxBytesInFrame int, lbls []mimirpb.LabelAdapter) int {
+func initializedFrameBytesRemaining(maxBytesInFrame int, lbls []prompb.Label) int {
 	frameBytesLeft := maxBytesInFrame
 	for _, lbl := range lbls {
 		frameBytesLeft -= lbl.Size()
 	}
 	return frameBytesLeft
 }
+
+// TODO what's the right package for this utility?
```

Comment:
It's pretty specific on what information it gets from the query related to this file, so I'd be ok with keeping it here and relying on the unit tests of is callers.

Otherwise the closes location and file to me seems pkg/util/labels.go  , so you could add pkg/util/remote.go and add it there. Although I don't know what a good name for it would be.

Reply:
Agree it's specific. I kept it here for now.